lib/trino-filesystem/src/main/java/io/trino/filesystem/memory/MemoryFileSystemCacheConfig.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.memory;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import io.airlift.units.MaxDataSize;
+import jakarta.validation.constraints.NotNull;
+
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.util.concurrent.TimeUnit.HOURS;
+
+public class MemoryFileSystemCacheConfig
+{
+    // Runtime.getRuntime().maxMemory() is not 100% stable and may return slightly different value over JVM lifetime. We use
+    // constant so default configuration for cache size is stable.
+    @VisibleForTesting
+    static final DataSize DEFAULT_CACHE_SIZE = DataSize.succinctBytes(Math.min(
+            Math.floorDiv(Runtime.getRuntime().maxMemory(), 10L),
+            DataSize.of(200, MEGABYTE).toBytes()));
```

Comment:
Why math.min?

if someone runs trino with -Xmx1g (eg for tests purposes, testcontainers, etc.) then alloting 200 MB to iceberg cache looks like a lot.

Reply:
is this allocation reported via `getInitialMemoryRequirement`?

Reply:
> Why math.min?
> if someone runs trino with -Xmx1g (eg for tests purposes, testcontainers, etc.) then alloting 200 MB to iceberg cache looks like a lot.

It's minimum of 10% of JVM max heap size and 200MB, so in your scenario it would be a 100mb cache. I can dial it down further to 5% if you think that's safer.

Reply:
> is this allocation reported via getInitialMemoryRequirement?

First time I heard about it's existence. There are zero implementations of it. The API docs say `initial memory requirement`, all connector caches are lazily populated, so it's unclear to me if this fits in the intended usage.
Even if this was reported, it just tries to reserve memory in `LocalMemoryManager#memoryPool`. That would help on worker nodes, but I don't see how that helps on coordinator (where this cache lives) as we don't track coordinator memory usage in memory pool today.